### DISCUSSION TITLE
<Link relative="path"> behaviour with v2 route convention

### DISCUSSION THREAD

**Discussion by bairamau** [remix-run/remix#6230] [Q&A] [answered]
Is there a usecase for `Link` component with `relative="path"` prop with routes v2?

The example in [docs](https://reactrouter.com/en/main/components/link#relative) is shown for nested `Route` components, how does it translate to file based routing in Remix?
We have a couple links with that prop in our codebase but I can't see the difference in behavior with or without it

**Comment by kiliman** [comment, marked as answer]
It depends on your routes. By default, `Link to=".."` will navigate up your *route* hierarchy. It will navigate URL segments with `Link to=".." relative="path"`. Most of the time, they end up being the same thing. However, some routes will include multiple path segments.

```
users.tsx => (layout)
users._index.tsx => /users list
users.$userId.tsx => /users/:userId
users.$userId_.edit.tsx => /users/:userId/edit (layout is users.tsx, note the trailing _)
```
```ts
<Routes>
  <Route file="root.tsx">
    <Route path="users" file="routes/users.tsx">
      <Route path=":userId" file="routes/users.$userId.tsx" />
      <Route path=":userId/edit" file="routes/users.$userId_.edit.tsx" />
      <Route index file="routes/users._index.tsx" />
    </Route>
  </Route>
</Routes>
```
If you were at `/users/123/edit`, `Link to=".."` will behave differently based on what `relative` is.

| `relative` | result |
|--------|--------|
| _default_ | /users |
| path | /users/123 | 

The edit route includes 2 URL segments but is treated as a _single_ route component. I hope this clarifies things.
